helm/kubernetes-nginx-ingress-controller-chart/templates/controller-hpa.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: autoscaling/v2beta1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Values.controller.name }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    app: {{ .Values.controller.name }}
+    giantswarm.io/service-type: "managed"
+    k8s-app: {{ .Values.controller.k8sAppLabel }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ .Values.controller.name }}
+  minReplicas: 2
+  # This is the maximum number of nodes we allow currently. Having more controllers than nodes should not happen.
+  maxReplicas: 100
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        targetAverageUtilization: 50
+    - type: Resource
+      resource:
+        name: memory
+        targetAverageUtilization: 50
```

Comment:
Where did this value come from?

Reply:
I checked some other repos and found the same value here: https://gitlab.com/charts/gitlab/blob/master/charts/nginx/values.yaml#L155-156

Also @puja108 checked with some contacts who also recommended 50%. We should see how this works out though.

Reply:
Manuel, the maintainer of the IC said (paraphrased) "nginx above 50% is a strong sign for high load because of the way nginx works, you might waste a bit of resources but it's more safe that way"

IMO it is still most probably less resources than what we have now

Reply:
for  CPU that make sense to me, but for memory not so sure

Reply:
hmm, right, definitely needs to be tuned separately, and tested

Reply:
Yeah, as Vaclav said, 50% of CPU utilization but not memory.